mikeio1d/result_network/geometry.py

```diff
@@ -0,0 +1,238 @@
+            res1d_reaches = [res1d_reaches]
+
+        points = []
+        for reach in res1d_reaches:
+            points.extend([ReachPoint.from_digipoint(dp) for dp in reach.DigiPoints])
+            points.extend([ReachPoint.from_gridpoint(gp) for gp in reach.GridPoints])
+
+        return ReachGeometry(points)
+
+    @property
+    def chainages(self) -> List[float]:
+        return [p.chainage for p in self._get_unique_points()]
+
+    @property
+    def points(self) -> List[ReachPoint]:
+        return self._points
+
+    @property
+    def digipoints(self) -> List[ReachPoint]:
+        return [p for p in self.points if p.is_digipoint()]
+
+    @property
+    def gridpoints(self) -> List[ReachPoint]:
+        return [p for p in self.points if p.is_gridpoint()]
+
+    @property
+    def length(self) -> float:
+        return self.chainages[-1] - self.chainages[0]
+
+    def to_shapely(self) -> BaseGeometry:
+        """Convert to a shapely geometry."""
+        points = self._get_unique_points()
+        xy = [(p.x, p.y) for p in points]
+        return LineString(xy)
+
+    def chainage_to_geometric_distance(self, chainage: float) -> float:
+        """Convert chainage to geometric distance."""
+        chainages = self.chainages
+        distances = self._get_distances()
+        if chainage < chainages[0] or chainage > chainages[-1]:
+            raise ValueError(
+                f"Chainage of {chainage} outside reach range of {chainages[0]} to {chainages[-1]}"
+            )
+        distance = float(np.interp(chainage, chainages, distances))
+        return distance
+
+    def chainage_from_geometric_distance(self, geometric_distance: float) -> float:
+        """Convert geometric distance to chainage."""
+        chainages = self.chainages
+        distances = self._get_distances()
+        if geometric_distance < distances[0] or geometric_distance > distances[-1]:
+            raise ValueError(
+                f"Geometric distance of {geometric_distance} outside reach range of {distances[0]} to {distances[-1]}"
+            )
+        chainage = float(np.interp(geometric_distance, distances, chainages))
+        return chainage
+
+    @lru_cache(maxsize=None)
+    def _get_unique_points(self) -> List[ReachPoint]:
+        """Removes points sharing the same chainage and coordinates."""
+        return sorted(list(set(self._points)))
+
+    @lru_cache(maxsize=None)
+    def _get_distances(self) -> List[float]:
+        """Returns a list of geometric distances between all unique points."""
+        points = self._get_unique_points()
+        distances = []
+        total_distance = 0.0
+        prev_point = points[0]
+        for point in points:
+            distance = point.to_shapely().distance(prev_point.to_shapely())
+            total_distance += distance
+            distances.append(total_distance)
+            prev_point = point
+        return distances
+
+
+@dataclass(frozen=True)
+class CatchmentGeometry:
```

Comment:
I know that in Python it is accepted to dump a lot of things into a single file not to deal with module resolution when classes are needed. Still I suggest to split the contents of geometry.py file into separate file, because it is easier to navigate. I think we could even move the geomtry.py into separate folder like mikeio1d/geometry and have a single file per class, etc.